### PR TITLE
Most fixes for Windows and MSVC 2017

### PR DESCRIPTION
**Pull request by acterhd**
Here is some workarounds for Windows. 
- Renamed literal logic ops to symbolic
- Fixed some compatibility issues
- Disabled checking for CXX14 (WHAT?)

### CODE REVIEW

**Comment by hhirsch** [comment]
Why isn't this a PR to the vs branch? The vs branch should be way closer to what you need.

The log files and commented out lines need to go.

Adding Dorians points here:

1st we should indeed strip references to epoxy and make a loader-neutral include
2nd replacing runtime variables by constants is absolutely unacceptable
like the 45 GL constant
or forcing uniform buffers on
3rd, don't include standard headers in the project, even less in the public "include" folder (getopt being an "exception" somehow)
4th, don't let bad submodule refs be committed
(see gwen)
5th, don't disable Radix dependency loading in the root CMakeFile
6th, don't use "-D<thing>", use the dedicated cmake function for that
7th don't define it for all platforms
8th --std=c++1z should be in FindCXX14.cmake as a flag candidate
Also I'd not commit the generated glad file

**Comment by SGOrava** [comment]
not, and, or should stay
if your compiler does not support them request their support....
our test for C++14 works on Linux and Mac and even with mingw on Windows, if you want to add some platform / compiler specific changes, please test it with GCC too and do it conditionaly in cmake....

**Comment by acterhd** [comment]
Sorry, no....

**Comment by acterhd** [comment]
But how to say.... problem with buggy generator

File: source/renderer/UiRenderer.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ void UiRenderer::render() {
   Text version;
   version.font     = "Pacaya";
   version.size     = 0.5f;
-  version.content  = std::string("Early testing build: ")  + GAME_VERSION;
+  version.content  = std::string("Early testing build: ") + "development 0.4.0-180-g3805388";
```
Comment (SGOrava) [inline source/renderer/UiRenderer.cpp]: GAME_VERSION is saved in file Version.hpp and this file is regenerated with cmake in each build

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,11 +9,24 @@ project(${EXE_NAME})
...
+#### PLATFORM-SPECIFIC FIXES ####
+
+if(DEFINED WIN32) # Support for Windows based
+    add_definitions("-DNOMINMAX")
+endif()
+
+if(DEFINED WIN32) # Workaround for MSVC (and, or, not fix)
+    add_definitions("-Dnot=!")
+    add_definitions("-Dor=||")
+    add_definitions("-Dand=&&")
```
Comment (SGOrava) [inline CMakeLists.txt]: How does this work with mingw gcc ?

**Pull request closed without merging**